### PR TITLE
Story/RWA-1057 - adds TOIL to SDK

### PR DESCRIPTION
**Pull request by HG-rc**
Ticket: https://rotacloud.atlassian.net/browse/RWA-1057

Adds methods for fetching data for:
TOIL accruals
TOIL allowances.

Delete for: TOIL accruals
Post fro: TOIL accruals

NB// deleting/creating won't be functional until the API for TOIL is no longer using dummy data.

### CODE REVIEW

File: src/models/toil-accrual.model.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  public created_by: number;
+  public date: string;
+  public deleted: boolean;
+  public deleted_at?: string;
+  public deleted_by?: string;
+  public duration_hours: number;
+  public id: number;
+  public leave_year: number;
+  public location_id: number;
+  public user_id: number;
```
Comment (liammoore4) [inline src/models/toil-accrual.model.ts:15]: Super minor as I think we will drop these model classes at some point in the future but is the `public` keyword used on other model classes? I'd be in favour of dropping these but if they are present on other model classes, I'm happy to leave them

File: src/services/toil-accruals.service.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  }
+
+  async *list(query: ToilAccrualsQueryParams, options?: Options) {
+    for await (const res of super.iterator<ApiToilAccrual>({ url: this.apiPath, params: query }, options)) {
+      yield new ToilAccrual(res);
+    }
+  }
+
+  listAll(query: ToilAccrualsQueryParams, options?: Options): Promise<ToilAccrual[]>;
+  async listAll(query: ToilAccrualsQueryParams, options?: Options) {
```
Comment (liammoore4) [inline src/services/toil-accruals.service.ts:40]: Unecessary overload?
```suggestion
  async listAll(query: ToilAccrualsQueryParams, options?: Options): Promise<ToilAccrual[]> {
```

File: src/services/toil-allowance.service.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    for await (const res of super.iterator<ApiToilAllowance>(
+      { url: `${this.apiPath}/${year}`, params: query },
+      options
+    )) {
+      yield new ToilAllowance(res);
+    }
+  }
+
+  listAll(year: number, query: ToilAllowanceQueryParams, options?: Options): Promise<ToilAllowance[]>;
+  async listAll(year: number, query: ToilAllowanceQueryParams, options?: Options) {
```
Comment (liammoore4) [inline src/services/toil-allowance.service.ts:19]: Same as above

File: src/models/toil-accrual.model.ts (hunk 1 above)
Comment (HG-rc) [reply src/models/toil-accrual.model.ts:15]: They do all have them

File: src/services/toil-accruals.service.ts (hunk 2 above)
Comment (HG-rc) [reply src/services/toil-accruals.service.ts:40]: good shout have removed, these do match those from the other services - any idea why they might have been needed?

Comment (liammoore4) [reply src/services/toil-accruals.service.ts]: Oh! Not sure doesn't look like they are needed. Guessing it was a half complete overload for the `rawResponse` maybe?  🤷‍♂️

**Review by liammoore4 — APPROVED** [review]

**Pull request merged**